@xen-orchestra/web-core/lib/layouts/CoreLayout.vue

```diff
@@ -0,0 +1,184 @@
+      <slot name="app-logo" />
+      <UiButtonIcon
+        v-tooltip="sidebarStore.isExpanded ? 'Close sidebar' : 'Open sidebar'"
+        :icon="sidebarStore.isExpanded ? faAngleDoubleLeft : faBars"
+        class="sidebar-toggle"
+        @click="sidebarStore.toggleExpand()"
+      />
+      <slot name="app-header" />
+    </header>
+    <div class="container">
+      <div
+        v-if="sidebarStore.isExpanded && !sidebarStore.isPinned"
+        class="sidebar-overlay"
+        @click="sidebarStore.toggleExpand(false)"
+      />
+      <LayoutSidebar class="sidebar">
+        <template #header>
+          <slot name="sidebar-header" />
+        </template>
+        <template #default>
+          <slot name="sidebar-content" />
+        </template>
+        <template #footer>
+          <slot name="sidebar-footer" />
+        </template>
+      </LayoutSidebar>
+      <div class="main-container">
+        <header>
+          <slot name="content-header" />
+        </header>
+        <main class="main">
+          <div class="content">
+            <slot name="content" />
+          </div>
+          <div v-if="isPanelVisible" :class="{ mobile: uiStore.isMobile }" class="panel">
+            <header v-if="$slots['panel-header'] || uiStore.isMobile" class="panel-header">
+              <UiButtonIcon
+                v-if="uiStore.isMobile"
+                :icon="faAngleLeft"
+                class="panel-close-icon"
+                @click="panelStore.close()"
+              />
+              <slot name="panel-header" />
+            </header>
+            <div v-if="$slots['panel-content']" class="panel-content">
+              <slot name="panel-content" />
+            </div>
+          </div>
+        </main>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import UiButtonIcon from '@core/components/button/ButtonIcon.vue'
+import LayoutSidebar from '@core/components/layout/LayoutSidebar.vue'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import { usePanelStore } from '@core/stores/panel.store'
+import { useSidebarStore } from '@core/stores/sidebar.store'
+import { useUiStore } from '@core/stores/ui.store'
+import { faAngleDoubleLeft, faAngleLeft, faBars } from '@fortawesome/free-solid-svg-icons'
+import { computed } from 'vue'
+
+const sidebarStore = useSidebarStore()
+const panelStore = usePanelStore()
+const uiStore = useUiStore()
+
+const slots = defineSlots<{
+  'app-logo'(): any
+  'app-header'(): any
+  'sidebar-header'(): any
+  'sidebar-content'(): any
+  'sidebar-footer'(): any
+  'content-header'(): any
+  content(): any
+  'panel-header'(): any
+  'panel-content'(): any
+}>()
+
+const isPanelVisible = computed(() => {
+  if (!slots['panel-header'] && !slots['panel-content']) {
+    return false
+  }
+
+  return panelStore.isExpanded
+})
+</script>
+
+<style lang="postcss" scoped>
+.sidebar-overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
```

Comment:
What do you think of replacing these lines with `inset`?
```suggestion
  inset: 0;
```